src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -101,16 +117,84 @@ export function ActiveWeb3Provider({
     activating = updateContext.activating ?? false
     active = updateContext.active ?? false
     chainId = updateContext.chainId
-    error = updateContext.error
     library = updateContext.library as Web3Provider
   }
 
-  // Log web3 errors to facilitate debugging.
-  useEffect(() => {
-    if (error) {
-      console.error('web3 error:', error)
+  return <Web3Context.Provider value={{ web3, updateWeb3 }}>{children}</Web3Context.Provider>
+}
+
+export type Web3Connection = [Connector, Web3ReactHooks]
+
+function toWeb3Connection<T extends Connector>([connector, hooks]: [T, Web3ReactHooks, Web3ReactStore]): [
+  T,
+  Web3ReactHooks
+] {
+  return [connector, hooks]
+}
+
+function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
+  // TODO(kristiehuang): implement RPC URL fallback, then jsonRpcEndpoint can be optional
+
+  // WalletConnect relies on Buffer, so it must be polyfilled.
+  if (!('Buffer' in window)) {
+    window.Buffer = Buffer
+  }
```

Comment:
![image](https://user-images.githubusercontent.com/5773490/176786881-d678d769-47fa-47a1-a5c5-fb623a73503d.png)

I think you should add this to the global window type

Reply:
I'm not getting this warning - are you using `"@types/node": "^16.7.13"` ?

Reply:
> > Should merging be deferred until [NoahZinsmeister/web3-react#591](https://github.com/NoahZinsmeister/web3-react/issues/591) is resolved, so that this doesn't bloat the widget size?
> 
> is there a plan for this to be resolved? I don't want to be blocked by an issue that's on the backlog

We've gotten pushback from integrators in the past about the size of this package, so I would investigate this before moving forward. It shouldn't block this PR, but it may need to block wallet-connect-flow from merging into main, or it may need to be noted in the release notes.